### ISSUE TITLE
android token issue

### ISSUE THREAD

**Issue by muhammadsarim555** [watanabeyu/react-native-simple-twitter#23]
hi, i am using your package . if i am using in the ios this is working fine further more when i use android it is  not giving any token and redirect

**Comment by mark-muhammad** [comment]
@muhammadsarim555 i have the same problem, have you found any solution?

**Comment by muhammadsarim555** [comment]
@mark-muhammad  i did not find this solution

**Comment by watanabeyu** [comment]
Hi there!
I will check this, so please wait!
sorry.

FYI, please let me know your information.
- OS
- Device
- react-native-simpel-twitter version
- React Native version

**Comment by muhammadsarim555** [comment]
@watanabeyu  WIndow 7 , expo sdk version 31 , device android LG

**Comment by mark-muhammad** [comment]
Mine is slightly different
- OS: macOS 10.14
- Device: android 5.1.1
- react native: 31.0.0
- react-native-simple-twitter: 2.0.2

Also, I've found that in android, callback URL was not called.

**Comment by watanabeyu** [comment]
@mark-muhammad 
I check it on below environment.

- OS: Android 7.0
- Device: HUAWEI P10 lite
- react-native-simpel-twitter version: 2.2.0
- React Native version(expo):v32

 I can work fine.
Please check your environment working at this [example](https://github.com/watanabeyu/react-native-simple-twitter/tree/master/example).

**Comment by mark-muhammad** [comment]
@watanabeyu I did follow the example but still didn't work. Do you think the version I used could be the problem? If I really should update the lib, what version do you think still compatible with react native v31? Because I once used v32 but got a problem which forced me to use v31 instead.

**Comment by watanabeyu** [comment]
@mark-muhammad 
umm... My Android phone is work fine.
Could you show me the source code?

**Comment by mark-muhammad** [comment]
@watanabeyu I just updated to v2.3.0 and still didn't work. Anyway, here is my code related to the login process. What made curious is that it worked fine in iOS.

```
onGetAccessToken = ({ oauth_token, oauth_token_secret }) => {
    console.log("TOKEN_SET: " + oauth_token + " | " + oauth_token_secret)
}

onSuccess = (user) => {
    console.log(user);
    TwitterID = user.id;
    Name = user.name;
    Email = user.email;

    this.loadAccessKeysViaTwtitter(TwitterID, Name, Email);
}

onClose = (e) => {
    console.log("press close button")
}

onError = (err) => {
    console.log(err)
}

<TWLoginButton 
    containerStyle={styles.loginContainer}
    style={styles.loginButton}
    textStyle={styles.loginButtonText}
    onGetAccessToken={this.onGetAccessToken}
    onSuccess={this.onSuccess}
    closeTextStyle={styles.loginCloseText}
    onClose={this.onClose}
    onError={this.onError}>
    <SocialIcon
        title='TWITTER LOGIN'
        button
        iconSize={ 18 }
        iconStyle={{ marginTop: -8, marginBottom: -8 }}
        fontStyle={{ fontSize: 12 }}
        style={{width: '90%', height: 40, borderRadius: 2, alignSelf: 'center'}}
        type='twitter'
    />
</TWLoginButton>```

**Comment by mark-muhammad** [comment]
@watanabeyu I recently found out that my callback was actually hit withquery string `oauth_verifier` which you used to filter here 
`const match = webViewState.url.match(/\?oauth_token=.+&oauth_verifier=(.+)/);`

But somehow, `WebView` didn't read the URL, and since I have redirection in the callback, `WebView` recognized only the redirected URL, ignoring callback URL. Do you think it is due to how `WebView` in android handles redirection? Do you I should change the logic in the callback URL?

**Comment by mark-muhammad** [comment]
@watanabeyu Never mind. I just resend the query string which twitter sent to my callback URL to the redirection inside my callback. In short, in my callback URL, I did this
`redirect($url . $_SERVER['QUERY_STRING']);`

It's probably just a workaround, but if `WebView` in android only reads the redirected URL, I guess this is the way to go.
@muhammadsarim555 Could this happen to be your case too?

By the way, thank you for the good lib @watanabeyu

**Comment by muhammadsarim555** [comment]
@mark-muhammad  i did not show the  **web view. ** i was just unable to get token

**Comment by watanabeyu** [comment]
@mark-muhammad @muhammadsarim555 
Sorry to be late to response.
I fixed webview bug, and update package v2.4.0

please use this version.
